src/Microsoft.Azure.SignalR.Common/REST/GeneratedRestClient.cs

```diff
@@ -0,0 +1,36 @@
+
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.Rest;
+
+namespace Microsoft.Azure.SignalR
+{
+    public partial class GeneratedRestClient
+    {
+        private readonly string _productInfo;
+
+        public GeneratedRestClient(Uri baseUri, ServiceClientCredentials credentials, string productInfo) : this(baseUri, credentials)
+        {
+            _productInfo = productInfo;
+        }
+
+        partial void CustomInitialize()
+        {
+            HttpClient.DefaultRequestHeaders.Add(Constants.AsrsUserAgent, _productInfo);
+        }
+
+        public static GeneratedRestClient Build(string connectionString, string productInfo)
```

Comment:
use `_productInfo`? remove `productInfo`?

Reply:
This is a static method and productInfo will be passed to _productInfo?

Reply:
Your goal is to `HttpClient.DefaultRequestHeaders.Add(Constants.AsrsUserAgent, _productInfo);`, is `HttpClient` static, no need to set it every time the `GeneratedRestClient` is created?

Reply:
HttpClient is not a static field. It belongs to the base class of GeneratedRestClient.